server/storage/tuf_store.go

```diff
@@ -0,0 +1,115 @@
+package storage
+
+import (
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"github.com/docker/go/canonical/json"
+	"github.com/docker/notary"
+	"github.com/docker/notary/storage"
+	"github.com/docker/notary/tuf/data"
+)
+
+// TUFMetaStorage wraps a MetaStore in order to walk the TUF tree for GetCurrent in a consistent manner,
+// by always starting from a current timestamp and then looking up other data by hash
+type TUFMetaStorage struct {
+	MetaStore
+	// cached metadata by checksum
+	cachedMeta map[string]*storedMeta
```

Comment:
Non-blocking:  could we also use this cache to speed up a `GetChecksum` implementation for `TUFMetaStorage`?